modules/afmongodb/afmongodb-worker.c

```diff
@@ -29,15 +29,79 @@
 #include "value-pairs/value-pairs.h"
 #include "scanner/list-scanner/list-scanner.h"
 
+static LogThreadedResult _do_bulk_flush(MongoDBDestWorker *self);
+
+static void
+_compose_bulk_op_options(MongoDBDestWorker *self)
+{
+  MongoDBDestDriver *owner = (MongoDBDestDriver *) self->super.owner;
+
+  if (owner->use_bulk)
+    {
+      self->bson_opts = g_new0(bson_t, 1);
+      bson_t def_opts = BSON_INITIALIZER;
+      *self->bson_opts = def_opts;
+
+      if (FALSE == BSON_APPEND_BOOL(self->bson_opts, "ordered", false == owner->bulk_unordered))
+        msg_error("Error seting bulk option",
+                  evt_tag_str("option", "ordered"),
+                  evt_tag_str("driver", owner->super.super.super.id));
+
+      if (self->write_concern)
```

Comment:
`write_concern` is never null, this should probably be a g_assert().

Reply:
_compose_write_concern will crash already if it is cannot be allocated, removing instead